### PR TITLE
debug: Integrate automatic RenderDoc capturing.

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Signed-off-by: Hans-Kristian Arntzen <post@arntzen-software.no>

### CODE REVIEW

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -4736,6 +4744,38 @@ static void STDMETHODCALLTYPE d3d12_command_list_OMSetStencilRef(d3d12_command_l
...
+                {
+                    WARN("Triggering RenderDoc capture for this command list.\n");
+                    list->debug_capture = true;
+                    break;
+                }
+            }
+        }
+    }
+}
+#endif
```
Comment (doitsujin) [inline libs/vkd3d/command.c:4777]: Not a huge fan of having entire renderdoc-specific functions in common code; can these be moved to renderdoc.c instead?

File: libs/vkd3d/command.c [hunk 2]
```diff
@@ -7689,8 +7737,56 @@ static void d3d12_command_queue_signal(struct d3d12_command_queue *command_queue
...
+        capture_label.sType = VK_STRUCTURE_TYPE_DEBUG_UTILS_LABEL_EXT;
+        capture_label.pNext = NULL;
+        capture_label.pLabelName = "capture-marker,end_capture";
+        memset(capture_label.color, 0, sizeof(capture_label.color));
+        VK_CALL(vkQueueInsertDebugUtilsLabelEXT(command_queue->vkd3d_queue->vk_queue, &capture_label));
+    }
+    else
+        vkd3d_renderdoc_end_capture(command_queue->device->vkd3d_instance->vk_instance);
+}
+#endif
```
Comment (doitsujin) [inline libs/vkd3d/command.c:7786]: Same here.

File: libs/vkd3d/command.c [hunk 3]
```diff
@@ -7730,9 +7826,21 @@ static void d3d12_command_queue_execute(struct d3d12_command_queue *command_queu
         return;
     }
 
+#ifdef VKD3D_ENABLE_RENDERDOC
+    if (debug_capture)
+        debug_capture = d3d12_command_queue_begin_capture(command_queue);
+#else
+    (void)debug_capture;
```
Comment (doitsujin) [inline libs/vkd3d/command.c:7833]: what is the purpose of this exactly?

**Review by doitsujin — COMMENTED** [review]
There's a bit more ifdef magic going on here than I'm comfortable with, not quite sure how much this can be improved though.

Otherwise LGTM

File: libs/vkd3d/command.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c]: Fixed.

File: libs/vkd3d/command.c (hunk 3 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c:7758]: Added comment.

**Review by doitsujin — APPROVED** [review]

**Pull request merged**